treq/auth.py

```diff
@@ -26,6 +182,112 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             method, uri, headers=headers, bodyProducer=bodyProducer)
 
 
+class _RequestDigestAuthenticationAgent(object):
+
+    digest_auth_cache = {}
```

Comment:
Yeah, it would appear a shared cache would reduce the amount of requests needed for how I'm using this from 6 to 4. I'm already formatting it with a shared auth like your "and share a cache." example in my automation script.

I'm using this with automation code that hits 3 separate URL's comprising 2 GET's and 1 POST that all can share the same credentials and cache.

I'm a little unsure of how I should go about refactoring this though, got any hints on where I should start?